scipy/signal/ltisys.py

```diff
@@ -500,6 +501,419 @@ def freqresp(self, w=None, n=10000):
+
+    """
+    def __new__(cls, *system):
+        """Handle object conversion if input is an instance of lti"""
+        if len(system) == 1:
+            if isinstance(system[0], ZerosPolesGain):
+                return copy.deepcopy(system[0])
+            if isinstance(system[0], lti):
+                return system[0].to_zpk()
+
+        # No special conversion needed
+        return super(ZerosPolesGain, cls).__new__(cls)
+
+    def __init__(self, *system):
+        """Initialize the zeros, poles, gain LTI system
+
+        Parameters
+        ----------
+        system : arguments
+            The following arguments are possible
+            * an instance of the lti class (StateSpace, TransferFunction,
+              ZerosPolesGain)
+            * (zeros, poles, gain)
+
+        """
+        # Conversion of lti instances is handled in __new__
+        if isinstance(system[0], lti):
+            return
+
+        super(ZerosPolesGain, self).__init__(self, *system)
+
+        self._zeros = None
+        self._poles = None
+        self._gain = None
+
+        self.zeros, self.poles, self.gain = system
+
+    def __repr__(self):
+        """Return representation of the ZerosPolesGain system"""
+        return '{0}(\n{1},\n{2},\n{3}\n)'.format(
+            self.__class__.__name__,
+            repr(self.zeros),
+            repr(self.poles),
+            repr(self.gain),
+            )
+
+    @property
+    def zeros(self):
+        return self._zeros
+
+    @zeros.setter
+    def zeros(self, zeros):
+        self._zeros = atleast_1d(zeros)
+
+        # Update dimensions
+        if len(self.zeros.shape) > 1:
+            self.outputs, self.inputs = self.zeros.shape
+        else:
+            self.outputs = 1
+            self.inputs = 1
+
+    @property
+    def poles(self):
+        return self._poles
+
+    @poles.setter
+    def poles(self, poles):
+        self._poles = atleast_1d(poles)
+
+    @property
+    def gain(self):
+        return self._gain
+
+    @gain.setter
+    def gain(self, gain):
+        self._gain = gain
+
+    def _copy(self, system):
+        """Copy the parameters of another ZerosPolesGain system
+
+        Parameters
+        ----------
+        system : instance of ZerosPolesGain
+            The zeros, poles gain system that is to be copied
+
+        """
+        self.poles = system.poles
+        self.zeros = system.zeros
+        self.gain = system.gain
+
+    def to_tf(self):
+        """Convert system representation to transfer function.
+
+        Returns
+        -------
+        sys : instance of TransferFunction
+            Transfer function of the current system
+
+        """
+        return TransferFunction(*zpk2tf(self.zeros, self.poles, self.gain))
+
+    def to_zpk(self):
+        """Convert system representation to zeros, poles, gain.
+
+        Returns
+        -------
+        sys : instance of ZerosPolesGain
+            The current system (self)
+
+        """
+        return self
```

Comment:
> It seems like to_zpk should return a copy for consistency with to_ss (i.e., if the result is modified the original shouldn't be in either case)

So I actually had it that way in the code at some point, but changed it back. Here is my reasoning (open to debate):
```sys = StateSpace(1,2,3,4)```
Then I agree that ```sys2 = StateSpace(sys)``` should be a copy of sys (which it is). However in my mind this is different from ```sys.to_ss()```, which seems more akin to numpy's ```asarray```, i.e. asking to write ```sys``` as a ```StateSpace``` object. If it already is I would expect nothing to change. 

This is of course just personal preference, but there is actually a pretty good usecase: For example, let's say you want to write a function that takes any linear system as an input, but your later code requries a ```StateSpace``` object. Then ```lti(*args).to_ss()``` would make sure that you end up with a ```StateSpace``` system, without having to make a copy if ```*args``` is in state space form.

I'm not sure this makes sense, I was pretty much just thinking about numpy's ```array``` handling as a reference (probably there is some usecase where you want something else). Maybe if we stick to the current format we should rename ```to_ss()``` to ```as_ss()```. What do you think?